monitoring/mongodb/alerts.yaml

```diff
@@ -139,3 +139,66 @@ groups:
+    for: 1h
+    labels:
+      severity: warning
+    annotations:
+      description: "MongoDB pod `{{ $labels.pod }}` has been stuck in the 'RECOVERING' state for the past hour. The instance may not be able to join the replica set if the platform ingests a large number of operations during this time. This alert is expected if the 'Resync a Data Services MongoDB Member' procedure has recently been executed."
+      summary: MongoDB node recovering for too long
+
+  - alert: MongoDbPodRecovering
+    expr: |
+      avg_over_time(mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*", rs_state="3"}[1d]) == 3
+    for: 1d
+    labels:
+      severity: critical
+    annotations:
+      description: "MongoDB pod `{{ $labels.pod }}` has been stuck in the 'RECOVERING' state for the past day. The instance may not be able to join the replica set if the platform ingests a large number of operations during this time."
+      summary: MongoDB node recovering for too long
+
+  - alert: MongoDbInvalidState
+    expr: |
+      avg_over_time(mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*", rs_state=~"6|8|10"}[1m]) > 0
+    for: 5m
+    labels:
+      severity: critical
+    annotations:
+      description: "MongoDB pod `{{ $labels.pod }}` is repeatedly or continuously in the `{{ $labels.rs_state }}` state. Immediate attention is required. You may collect an sosreport for further analysis."
+      summary: "MongoDB node in an invalid state: 6 (UNKNOWN), 8 (DOWN), 10 (REMOVED)"
+
+
+  - alert: MongoDbPodStartup2
+    expr: |
+      avg_over_time(mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*", rs_state="5"}[1h]) == 5
+    for: 1h
+    labels:
+      severity: warning
+    annotations:
+      description: "MongoDB pod `{{ $labels.pod }}` has been in the 'STARTUP2' state for an hour. Please ensure that the instance is running properly. You may follow the 'Resync a Data Services MongoDB Member' procedure to unblock this MongoDB instance instance."
+      summary: MongoDB node in STARTUP2 state for too long
+
+  - alert: MongoDbRSNotSynced
+    expr: |
+      sum by (rs_nm) (mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*", member_state="SECONDARY"}) != (${replicas} - 1)
+    for: 10m
+    labels:
+      severity: warning
+    annotations:
+      description: "MongoDB replica set `{{ $labels.rs_nm }}` is missing `{{ $value }}` SECONDARY members for the last 10 minutes. If the member is not recovering, you may collect an sosreport for further analysis."
+      summary: MongoDB replica set out of sync
+    
+  - alert: MongoDbRSNotProtected
+    expr: |
+      sum by (rs_nm) (mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*", member_state="SECONDARY"}) == 0
+      and on() (vector(${replicas}) > 1)
+    for: 10m
+    labels:
+      severity: critical
```

Comment:
not sure this extra level is useful.

In order of criticality:
* having no primary is critical
* not having a quorum is critical, but less. However, it is not so trivial to define when that happens in general case: esp. at 0, we are sure we don't have it... but it may be the case before we reach 0 ; and if there is no quorum this should raise other alerts (because there should be no primary)
* if we have the quorum, missing secondaries is such a warning that we are "less protected"

Reply:
The purpose of this alert is to flag a situation where all secondaries are missing, which, while not immediately catastrophic, significantly increases the risk of data loss in case of primary failure. It complements other alerts, such as quorum loss or primary failure, which would trigger more immediate critical alerts.
We could adjust the severity of this alert to a warning to better reflect the situation's urgency, while keeping the focus on redundancy and resilience of the replica set.

Reply:
it does not create more risk of data loss : if there is no quorum, write is not possible anymore --> this is only an issue of availability.

the redundancy risk is already covered by the previous alert, and the availability is covered by the alert on missing primary.

Reply:
> it does not create more risk of data loss : if there is no quorum, write is not possible anymore, and it is only an issue of availability.

Didn't we got cases where some secondaries were out of the replicaset configuration for some reason? In this case, we still have the quorum...

Reply:
yes, and we will see it with the "warning" alert above.